jax/numpy/linalg.py

```diff
@@ -136,6 +136,21 @@ def eigvalsh(a, UPLO='L'):
   return w
 
 
+@_wraps(onp.linalg.pinv)
+def pinv(a, rcond=1e-15):
+  a = np.conj(a)
+  rcond = np.asarray(rcond)
+  u, s, v = svd(a, full_matrices=False)
+  # Singular values less than or equal to ``rcond * largest_singular_value`` 
+  # are set to zero.
+  cutoff = rcond[..., np.newaxis] * np.amax(s, axis=-1, keepdims=True)
+  large = s > cutoff
+  s = np.divide(1, s)
+  s = np.where(large, s, 0)
+  res = np.matmul(np.transpose(v), np.multiply(s[..., np.newaxis], np.transpose(u)))
```

Comment:
Note that `transpose` function that NumPy uses here is not *quite* the same as `np.transpose` -- it only flips the last two axes. Is this version still correct?

Reply:
Good catch! in this case `np.swapaxes` would be more appropriate?